src/lib.rs

```diff
@@ -277,97 +208,111 @@ impl UPClientZenoh {
         };
         Ok(uattributes)
     }
+
+    /*        The table for mapping resource ID to message type
+     *
+     *  |   src rid   | sink rid | Publish | Notification | Request | Response |
+     *  |-------------|----------|---------|--------------|---------|----------|
+     *  | [8000-FFFF) |   None   |    V    |              |         |          |
+     *  | [8000-FFFF) |     0    |         |      V       |         |          |
+     *  |      0      | (0-8000) |         |              |    V    |          |
+     *  |   (0-8000)  |     0    |         |              |         |    V     |
+     *  |     FFFF    |     0    |         |      V       |         |    V     |
+     *  |     FFFF    | (0-8000) |         |              |    V    |          |
+     *  |      0      |   FFFF   |         |              |    V    |          |
+     *  |   (0-8000)  |   FFFF   |         |              |         |    V     |
+     *  | [8000-FFFF) |   FFFF   |         |      V       |         |          |
+     *  |     FFFF    |   FFFF   |         |      V       |    V    |    V     |
+     *
+     *  Some organization:
+     *  - Publish: {[8000-FFFF), None}
+     *  - Notification: {[8000-FFFF), 0}, {[8000-FFFF), FFFF]}, {FFFF, 0}, {FFFF, FFFF}
+     *  - Request: {0, (0-8000)}, {0, FFFF}, {FFFF, (0-8000)}, {FFFF, FFFF}
+     *  - Response: {(0-8000), 0}, {(0-8000), FFFF}, (FFFF, 0), {FFFF, FFFF}
+     */
+    #[allow(clippy::nonminimal_bool)] // Don't simplify the boolean expression for better understanding
+    fn get_listener_message_type(
```

Comment:
IMHO something very similar could be useful in up-rust with modifications to support the uStreamer case of all point to point messages from a certain authority.